src/Kernel32.Desktop/Kernel32.Helpers.cs

```diff
@@ -197,26 +197,27 @@ public static IEnumerable<MODULEENTRY32> Module32Enumerate(SafeObjectHandle hSna
             }
         }
 
-        public static string QueryFullProcessImageName(
+        public static unsafe string QueryFullProcessImageName(
             SafeObjectHandle hProcess,
             QueryFullProcessImageNameFlags dwFlags = QueryFullProcessImageNameFlags.None)
         {
             // If we ever resize over this value something got really wrong
             const int maximumRealisticSize = 1 * 1024 * 2014;
 
-            var buffer = new StringBuilder(255);
+            int size = 255;
+
             do
             {
-                int size = buffer.Capacity;
-                bool success = QueryFullProcessImageName(hProcess, dwFlags, buffer, ref size);
+                var text = stackalloc char[size];
```

Comment:
OK. I am not much versed on the pointer and unsafe memory management on .NET yet. I was just repeating the pattern that I learned to do with this project when I needed to pass on a string as `char*` to an unsafe API.

This is very interesting information that you brought us,  I don't follow your calculation of 3.5 Mo, is there any overhead/padding memory that I am not seeing here? Running through the progression: 255 + 255*2 + 255*4 + 255*8 + 255*16 + 255*32 ... it lead me to 2.8 Mo.

Anyway, may I use a simple `new char[size]` here? Isn't the managed array be incompatible with the unsafe pointer? Do I need to do some memory pinning or special Marshaling on this case? Sorry for all this questions, but you baffled me now ;-)